drivers/spi/spi_pl022.c

```diff
@@ -0,0 +1,334 @@
+					SPI_REG_ADDR(base, SPI_CPSR_OFFSET),
+					BYTE_MASK, CLK_PRESCALE_SHIFT, i);
+
+				/* Set optimal SCR value */
+				sys_set_bits(
+					SPI_REG_ADDR(base, SPI_CR0_OFFSET),
+					BYTE_MASK, SERIAL_CLKRATE_SHIFT, j - 1);
+				return;
+			}
+	/* Set div value to default if frequency is less*/
+	sys_set_bits(SPI_REG_ADDR(base, SPI_CPSR_OFFSET), 0xFF,
+		     CLK_PRESCALE_SHIFT, CLK_PRESCALE_MAX);
+
+	/* Set SCR value to default if frequency is less */
+	sys_set_bits(SPI_REG_ADDR(base, SPI_CR0_OFFSET), 0xFF,
+		     SERIAL_CLKRATE_SHIFT, SERIAL_CLKRATE_MAX);
+}
+
+static int spi_pl022_configure(struct device *dev,
+				 const struct spi_config *config)
+{
+	const struct spi_pl022_config *cfg = dev->config->config_info;
+	struct spi_pl022_data *data = dev->driver_data;
+	u32_t base = cfg->base;
+
+	if (spi_context_configured(&data->ctx, config)) {
+		/* Nothing to do */
+		return 0;
+	}
+
+	if (SPI_OP_MODE_GET(config->operation) != SPI_OP_MODE_MASTER) {
+		/* Slave mode is not implemented. */
+		return -ENOTSUP;
+	}
+
+	/* Setting SSP in Master Mode */
+	sys_clear_bit(SPI_REG_ADDR(base, SPI_CR1_OFFSET), MODE);
+
+	/* Setting Frame Format to SPI */
+	sys_set_bits(SPI_REG_ADDR(base, SPI_CR0_OFFSET),
+		     TWOBIT_MASK, FRF_SHIFT, FRF_SPI);
+
+	if ((config->operation & SPI_MODE_CPOL) != 0) {
+		/* Set CPOL if configured */
+		sys_set_bit(SPI_REG_ADDR(base, SPI_CR0_OFFSET), CPOL);
+	}
+
+	if ((config->operation & SPI_MODE_CPHA) != 0) {
+		/* Set CPHA if configured */
+		sys_set_bit(SPI_REG_ADDR(base, SPI_CR0_OFFSET), CPHA);
+	}
+
+	if ((config->operation & SPI_MODE_LOOP) != 0) {
+		/* Setting SSP to Loopback Mode */
+		sys_set_bit(SPI_REG_ADDR(base, SPI_CR1_OFFSET), LOOPBACK);
+	}
+
+	if (SPI_WORD_SIZE_GET(config->operation) != 8) {
+		return -ENOTSUP;
+	}
+
+	/* 8 bits frame per transfer */
+	sys_set_bits(SPI_REG_ADDR(base, SPI_CR0_OFFSET), NIBBLE_MASK,
+		     DSS_SHIFT, DSS_8BITS);
+
+	/* Setting the configured frequency */
+	set_frequency(dev, config->frequency);
+
+	/* At this point, it's mandatory to set this on the context! */
+	data->ctx.config = config;
+
+	spi_context_cs_configure(&data->ctx);
+
+	return 0;
+}
+
+static int spi_pl022_transceive(struct device *dev,
+				  const struct spi_config *config,
+				  const struct spi_buf_set *tx_bufs,
+				  const struct spi_buf_set *rx_bufs)
+{
+	const struct spi_pl022_config *cfg = dev->config->config_info;
+	struct spi_pl022_data *data = dev->driver_data;
+	u32_t base = cfg->base;
+	int err;
+
+	spi_context_lock(&data->ctx, false, NULL);
+
+	/* Wait for previous transfer complete */
+	wait_for_sync(base);
+
+	rx_buffer_flush(base);
+
+	err = spi_pl022_configure(dev, config);
+	if (err != 0) {
+		goto done;
+	}
+
+	spi_context_buffers_setup(&data->ctx, tx_bufs, rx_bufs, 1);
+
+	spi_context_cs_control(&data->ctx, true);
+
+	/* Set SSE bit, enable SSP */
+	sys_set_bit(SPI_REG_ADDR(base, SPI_CR1_OFFSET), SSE);
+
+	do {
+		push_data(dev);
+		pull_data(dev);
+	} while (spi_transfer_complete(dev));
```

Comment:
Btw, you don't have spi interrupt support on such controller?

Reply:
Added interrupt support